core/src/display_object/avm2_button.rs

```diff
@@ -643,6 +643,20 @@ impl<'gc> TDisplayObject<'gc> for Avm2Button<'gc> {
         BoundingBox::default()
     }
 
+    fn bounds_with_transform(&self, matrix: &Matrix) -> BoundingBox {
+        // Get self bounds
+        let mut bounds = self.self_bounds().transform(matrix);
+
+        // Add the bounds of the child, dictated by current state
+        let state = self.0.read().state;
+        if let Some(child) = self.get_state_child(state.into()) {
+            let child_bounds = child.bounds_with_transform(matrix);
+            bounds.union(&child_bounds);
+        }
+
+        bounds
+    }
+
     fn hit_test_shape(
```

Comment:
hit_test_shape has the same precondition that you just changed in mouse_pick.

> hit_area is not actually a child, so transform point into local space before passing it down.

should it be changed here too?

Reply:
Nice catch, it probably should